lib/internal/file.js

```diff
@@ -55,16 +66,24 @@ class File extends Blob {
       lastModified = DateNow();
     }
 
-    this.#name = toUSVString(fileName);
-    this.#lastModified = lastModified;
+    this[kState].name = toUSVString(fileName);
```

Comment:
I think we should keep the previous private properties. I didn't quite understand why you had to change them. Can you explain?

Reply:
afaict it's not possible to set private properties in kDeserialize. I spent way too long trying to find a way to make it work and nothing worked (using static blocks, binding kDeserialize, etc.).

Reply:
We are refactoring in the whole project to avoid using invalid_this and use private properties. I think we should not revert it.

Reply:
we can't fix this without using symbols

Reply:
@nodejs/buffer @TimothyGu any suggestions?

Reply:
You can do the following:

```js
class MyClass {
  #deserialize
  static setKDeserialize(instance, value) {
    instance.#deserialize = value;
  }
}
const {setKDeserialize} = MyClass;
delete MyClass.setKDeserialize;

setKDeserialize(new MyClass, 'hello');
```

Reply:
I tried with static init blocks already:

```js
let setKDeserialize

class File {
  #deserialize

  static {
    setKDeserialize = (o, value) => o.#deserialize = value
  }
}
```

Reply:
The issue is we don't get the serialized data we provide in kClone until kDeserialize is called. And setting the private values there throws an error, which makes sense. If we got the data in `ClonedFile` (which is the function I provide as the serialized info) we could just create a new File that way, but it's not called with any arguments or anything that can be used afaik.

Reply:
FWIW I tried with the following diff:

```diff
diff --git a/lib/internal/file.js b/lib/internal/file.js
index 396b7c5bf9..2487d8b12d 100644
--- a/lib/internal/file.js
+++ b/lib/internal/file.js
@@ -72,16 +72,15 @@ class File extends Blob {
     makeTransferable(this);
   }
 
+  #brand;
   get name() {
-    if (!isFile(this))
-      throw new ERR_INVALID_THIS('File');
+    this.#brand;
 
     return this[kState].name;
   }
 
   get lastModified() {
-    if (!isFile(this))
-      throw new ERR_INVALID_THIS('File');
+    this.#brand;
 
     return this[kState].lastModified;
   }
```

And even that wasn't working, as the cloned object is not constructed by the `File` constructor (and can't be AFAICT).